src/content/1.7/modules/concepts/services/_index.md

```diff
@@ -154,6 +161,82 @@ If you need more details about dependency injection and how services work in the
 their documentation about the [Service Container](https://symfony.com/doc/3.4/service_container.html).
 {{% /notice %}}
 
+### Override an existing Symfony service
+
+The container definition can be modified by a module, which enables you to override an existing Symfony service being used in PrestaShop.
+
+This is a mechanism similar to PrestaShop standard overrides, but the main benefit is that the php code stays unmodified. This prevents issues linked to code definition or autoloading failures.
+
+As you can read it from the [Symfony documentation](https://symfony.com/doc/current/service_container/service_decoration.html), there are 2 ways to modify an existing service:
+
+#### Override the service
+
+When you choose to override a service, this means that you _replace the service by another one_. The previous service is not usable anymore. Every other part of the code where this service is used will use the new version.
+
+To do it: you declare your new service using the old service name. So if you want to override the service `prestashop.core.b2b.b2b_feature` with your own implementation, you write in `config/services.yml` :
+
+```yml
+  prestashop.core.b2b.b2b_feature:
+    class: 'YourCompany\YourModule\YourService'
+```
+
+That's done. The service registered under the name `prestashop.core.b2b.b2b_feature` is now your service. The previous `prestashop.core.b2b.b2b_feature` is gone.
+
+#### Decorate the service
+
+When you choose to decorate a service, this means that you _make everybody use your service but you keep the old service available_. The previous service has been given a new name and can still be used. Every other part of the code where this service was used will use the new version.
+
+To do it: you declare your new service using the 'decorates' keyword. So if you want to decorates the service `prestashop.core.b2b.b2b_feature` with my own implementation, you write in `config/services.yml` :
+
+```yml
+ my_own_b2b_feature_service:
+    class: 'YourCompany\YourModule\YourService'
+    decorates: 'prestashop.core.b2b.b2b_feature'
+```
+
+That's done. The service registered under the name `prestashop.core.b2b.b2b_feature` is now your service. The previous `prestashop.core.b2b.b2b_feature` is still available under the name `prestashop.core.b2b.b2b_feature.inner`
+
+The decoration strategy can be very useful if:
+
+- you want some areas of your code to use the new service, some others to use the old service
+- you want to use the old service in the implementation of the new service
```

Comment:
But how do you force using the old service? With the inner name?
Then we agree that in all other places of the code, the new service will be used without the codebase being aware of it?

Reply:
Yes indeed, this is the Symfony behavior
So if you want to still use the old service, you need to modify the places where you want to use it and inject `[my-service].inner` instead